src/commonMain/kotlin/com.adamratzman.spotify/Builder.kt

```diff
@@ -311,21 +313,23 @@ class SpotifyClientApiBuilder(
         }
     }
 
-    override fun buildAsync(consumer: (SpotifyClientAPI) -> Unit) = Thread { consumer(build()) }.start()
+    override suspend fun buildAsync(consumer: (SpotifyClientApi) -> Unit) = coroutineScope {
+        withContext(Dispatchers.Default) { consumer(build()) }
+    }
```

Comment:
As I already said, this doesn't make any sense. This function blocks until the consumer finishes. I wouldn't call that async.